### ISSUE TITLE
promote search-list-box to racket/gui

### ISSUE THREAD

**Issue by spdegabrielle** [Metaxal/search-list-box#7]
In addition to being useful to user, having an official search-list-box control has the possibility to reduce some code duplication in the racket code-base; 
- the **Edit** menu’s **Show Active Keybindings** dialog 
- package manager dialog (pic below)

Other implementations of 'search-list-box'
- `qresults-list` https://github.com/alex-hhh/qresults-list

![image](https://user-images.githubusercontent.com/108047/145736887-8b83152d-a46a-4361-a414-aafff81d495e.png)

**Comment by spdegabrielle** [comment]
![image](https://user-images.githubusercontent.com/108047/145737480-4a4d31aa-6492-433d-9b89-289e3de538a7.png)

**Comment by Metaxal** [comment]
I agree it would be a good thing to have in racket/gui. I think I would need to upgrade it to use multiple columns though, at least.
